File: VendingMachine.Domain/CoinBank/PurchaseResult.cs
namespace VendingMachine.Domain.CoinBank;

public class PurchaseResult
{
    public int ChangeAmountInCents { get; set; }
    public int ActualSpentInCents { get; set; }
    public int PurchaseAmountInCents { get; set; }
    public Dictionary<int, int> UsedCoins { get; set; } = null!;
    public Dictionary<int, int> NewCoinBankState { get; set; } = null!;
    public Dictionary<int, int> ChangeCoins { get; set; } = null!;
}

File: vending-machine-client/src/app/+vending-machine/service/purchase-result.ts
import { CoinBank } from '../store/vending-machine.store';

export type PurchaseResult = {
  purchaseAmountInCents: number;
  actualSpentInCents: number;
  changeAmountInCents: number;
  usedCoins: CoinBank;
  newCoinBankState: CoinBank;
  changeCoins: CoinBank;
};
